src/game/movement.rs

```diff
@@ -25,9 +25,19 @@ pub(super) fn plugin(app: &mut App) {
     );
 }
 
+/// This is the input for the player's character controller,
+/// which in this case is only the direction the player wants to move in.
+///
+/// A character controller is the collection of systems that govern movement of characters.
+/// In our case, the character controller has the following logic:
+/// - Set the movement intent based on directional input.
+/// - Apply movement based on the intent.
+/// - Wrap the character within the window.
```

Comment:
Hmmm this seems more like module-level documentation. This component doesn't do all of those things.

Also, while this is technically only used as the player's character controller, it doesn't have to be strictly coupled to the player. Though it does currently happen to be, because the `record_movement_controller` system doesn't check for an additional marker component like `KeyboardMovementInput` or whatever.

Reply:
[`6877f00` (#230)](https://github.com/TheBevyFlock/bevy_quickstart/pull/230/commits/6877f00864851ed3edf5394c27efe8c143e79c66)

Reply:
(And subsequent)